Fix robots.txt 404 endpoint when individual_posts.enabled

hadronix opened the pull request:
Enhance individual post handling by updating routes with a post-type prefix to prevent errors with other .txt endpoints

hadronix committed:
Enhance individual post handling by updating routes with a post-type prefix to prevent errors with other .txt endpoints (such as robots.txt)

retlehs committed:
Merge branch 'main' into main

retlehs committed:
format

retlehs commented:
Is this working as expected for you? Individual posts `.txt` routes aren't working for me with this branch

hadronix committed:
fix individual posts route

hadronix commented:
Sorry, I unnecessarily changed the route while testing with multiple post types. The llms-txt command still outputs the regex variant so you can see at a glance which post types it's active for.

retlehs committed:
format

retlehs commented:
`post-slug.txt` still isn't working with that change

The sitemap generation, README, and code comments will also need updating to reflect the new pattern

hadronix committed:
fix individual posts urls in sitemap, update sitemap post fetcher

hadronix commented:
I can't recreate this problem; on a fresh install, it works, e.g., with posts and pages.
![acorn-test testpage-sample-page txt 2025-11-15 at 11 13 39 AM](https://github.com/user-attachments/assets/ab9610a0-ea9a-4ec0-8d3c-d4ed6971916b)

I added the correct url. there an other "problem" currently, the sitemap fetches all posts and within the ContentFetcher it just checks if the individual_posts are enabled, but it uses the main postTypes array from the config and not the postTypes from the individual_posts config part. I have not a complete overview of the code, for this reason i just added an filter before fetching the posts. this is only a simple suggestion, i don't thing this is the best solution. we cant just filter the posts because the postTypes from global config and individual may complete differ (i don't know if this would make sense a production environment).

I look forward to your feedback on this.

hadronix committed:
fix clearAllIndividualCaches
